# Видеокурс по теории вероятностей и математической статистике
# GeekBrains
## ДЗ Урока 1
## Казанцев Виталий

#### Задача 1

Имеется колода в 52 карты. Найти число возможностей вытянуть из неё 4 карты так, чтобы среди них был хотя бы один туз.

Задача аналогична примеру 6 из урока.
Сначала представим себе ситуацию когда выбирается один туз из колоды.
### $C_{4}^{1}=\frac{4!}{1!(4-1)!}=4$


In [1]:
import numpy as np
def combinations(n, k):
    return np.math.factorial(n) // (np.math.factorial(k) * np.math.factorial(n - k))
combinations(4, 1)

4

Остальные три карты выбираются из 48 карт.
### $С_{48}^{3}=\frac{48!}{3!(48-3)!}=17296$

In [2]:
combinations(48, 3)

17296

И в таком случае, когда из колоды общим числом в 52 карты выбрано 4 карты с одним тузом будет равно:
### $4\cdot17296=69184$   

In [3]:
a = combinations(4, 1)*combinations(48, 3)
a

69184

Теперь рассмотрим число возможностей на тот случай, кода среди выбранных карт не один, а два туза

In [4]:
b = combinations(4, 2)*combinations(48, 2)
b

6768

Cлучай, когда три туза

In [5]:
c = combinations(4, 3)*combinations(48, 1)
c

192

И случай, когда все четыре карты тузы.

In [6]:
d = combinations(4, 4)*combinations(48, 0)
d

1

Таким образом число возможностей, будет равно сумме этих возможностей

In [7]:
a+b+c+d

76145

Ответ: общее число возможностей вытянуть из колоды в 52 карты 4 карты так, чтобы среди них был хотя бы один туз, равно 76145

### Задачи 2 и 3

Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Случайным образом студент вытягивает 3 вопроса. Какова вероятность, что все выбранные вопросы знакомы студенту? Какова вероятность что два из трёх вопросов знакомы студенту?

### $\text{У нас есть два события. Первое событие } A \text{ студент вытянул билет, ответ на который он знает}$
### $\text{Второе событие } \overline A \text{ событие не-А, т.е. студент не занет ответ на вопрос в билете}$
Когда студент тянет первый билет, то вероятность того, что он вытянет билет с вопросм, ответ на который знаент равна:
### $A_1=\frac{5}{6}$ 
После этого события у нас остается 59 вопросов, из которых студент знает 49, а значит
### $A_2=\frac{49}{59}$
Теперь осталось 58 билетов, из которых 48 известны нашему студенту, значит:
### $A_3=\frac{48}{58}=\frac{24}{29}$
Первое, второе и третье событие независимы друг от друга. а значит вероятность наступления и $A_1$ и $A_2$ и $A_3$ равна произведению вероятностей этих событий, т.е
### $P=A_1\cdot A_2\cdot A_3=\frac{5\cdot 49\cdot 24}{6\cdot 59\cdot 29}\approx 0.57$

Ответ на тертий вопрос может зависить от того, каким билетом студент вытянет билет, ответ на который не знает. Если третьим, тогда вероятность $\overline A_3 = \frac{10}{58}=\frac{5}{29}$, тогда:
### $P_1=A_1\cdot A_2\cdot \overline A_3=\frac{5\cdot 49\cdot 5}{6\cdot 59\cdot 29}\approx 0.12$

Если незнакомый билет студент вытянет первым, тогда $\overline A_1=\frac{1}{6}$; $A_2=\frac{50}{59}$; $A_3=\frac{49}{58}$, тогда:
### $P_2=\overline A_1\cdot A_2\cdot A_3=\frac{1\cdot 50\cdot 49}{6\cdot 59\cdot 58}\approx 0.12$

Если незнакомы билет попадётся во вторую очередь, тогда $A_1=\frac{5}{6}$; $\overline A_2=\frac{10}{59}$; $A_3=\frac{49}{58}$, тогда:
### $P_3=\frac{5\cdot 10\cdot 49}{6\cdot59\cdot58}\approx 0.12$

Вероятность, что произойдет одно из трех событий равна
### $P=P_1+P_2+P_3\approx0.36$

Или другим способом.  
Общее число возможных комбинаций трех билетов из 60 $C_{60}^3=\frac{60!}{3!\cdot57!}$

In [8]:
a = combinations(60, 3)
a

34220

число благоприятных комбинаций, когда все три билета известны $C_{50}^3$

In [9]:
b = combinations(50, 3)
b

19600

ТОгда вероятность такого исхода

In [10]:
result_2 = b / a
result_2

0.5727644652250146

число возможных варианотов когда два билета известны, а один нет равна $C_{50}^2\cdotC_{10}^1$

In [11]:
c = combinations(50, 2)*combinations(10, 1)
c

12250

Тогда вероятность такого события

In [12]:
result_3 = c / a
result_3

0.35797779076563413

Ответ: У студента около 57% шанс на то, что все билеты будут ему знакомы,  и около 36%, что попадется один неизвестный билет

### Задача 4

Допустим, имеется некоторая очень редкая болезнь (поражает 0.1% населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99% больных этой болезнью и всего лишь в 1% случаев даёт ложный положительный ответ.

Вопрос: какова вероятность, что вы действительно больны ей?

Подсказка: используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.

$$P(B\:|\:A) = \frac{P(A\:|\:B) \cdot P(B)}{P(A)}$$

Пусть событие $B_1$ означает, что я болен и равно 0.001, а событие $B_2$ означает, что я не болен и равно 0.999. Событие А будет означать положительный тест, который верен в 99% случаев.  
Тогда вероятность того, что я болен при условии, что тест верный будет расчитываться по формуле:
### $P(B_1|A)=\frac{P(A|B_1)\cdot P(B_1)}{P(A)}$

Кроме того нам известно, что:
### $P(A|B_1)=0.99$
### $P(A|B_2)=0.01$
Тогда:
### $P(A)=B_1\cdot(A|B_1)+B_2\cdot(A|B_2)=0.001\cdot0.99+0.999\cdot0.01 =0.01098$

Получается:
### $P(B_1|A)=\frac{0.99\cdot0.001}{0.01098}\approx0,09$

Ответ: Вероятность того, что я имею заболевание, которым болеют 1 человек из тысячи, при условии что тест дал положительный результат равна примерно 9%.